`_SCHEMA` deprecations blog post (#18)
esphome/developers.esphome.io: docs/blog/posts/2025-05-14-schema-deprecations.md

---
date: 2025-05-14
authors: 
  - jesserockz
---

# `*.*_SCHEMA` deprecations

> In order to align all of the top level platform components (listed below), we are deprecating the `*_SCHEMA` constants that are present e.g., `SENSOR_SCHEMA`, `SWITCH_SCHEMA`, etc.

Reviewer: I've been avoiding use of "e.g.", "i.e." and the like after Gabriela mentioned to me that we don't use them in the HA docs -- it can be confusing for non-native English speakers.
```suggestion
In order to align all of the top level platform components (listed below), we are deprecating the `*_SCHEMA` constants that are present. Some examples are `SENSOR_SCHEMA`, `SWITCH_SCHEMA` and so on.
```
